qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+            **metadata: Union[configuration.Kernel,
+                              configuration.Discriminator]):
+    """Acquire for a ``duration`` on a ``channel`` and store the result
+    in a ``register``.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.MeasureChannel(0)
+        mem0 = pulse.MemorySlot(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.acquire(100, d0, mem0)
+
+            # measurement metadata
+            kernel = pulse.configuration.Kernel('linear_discriminator')
+            pulse.acquire(100, d0, mem0, kernel=kernel)
+
+    .. note:: The type of data acquire will depend on the execution
+        ``meas_level``.
+
+    Args:
+        duration: Duration to acquire data for
+        qubit_or_channel: Either the qubit to acquire data for or the specific
+            :class:`~qiskit.pulse.channels.AcquireChannel` to acquire on.
+        register: Location to store measured result.
+        metadata: Additional metadata for measurement. See
+            :class:`~qiskit.pulse.instructions.Acquire` for more information.
+
+    Raises:
+        exceptions.PulseError: If the register type is not supported.
+    """
+    if isinstance(qubit_or_channel, int):
+        qubit_or_channel = channels.AcquireChannel(qubit_or_channel)
+
+    if isinstance(register, channels.MemorySlot):
+        append_instruction(instructions.Acquire(
+            duration, qubit_or_channel, mem_slot=register, **metadata))
+    elif isinstance(register, channels.RegisterSlot):
+        append_instruction(instructions.Acquire(
+            duration, qubit_or_channel, reg_slot=register, **metadata))
+    else:
+        raise exceptions.PulseError(
+            'Register of type: "{}" is not supported'.format(type(register)))
+
+
+def set_frequency(frequency: float,
+                  channel: channels.PulseChannel):
+    """Set the ``frequency`` of a pulse ``channel``.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.set_frequency(1e9, d0)
+
+    Args:
+        frequency: Frequency in Hz to set channel to.
+        channel: Channel to set frequency of.
+    """
+    append_instruction(instructions.SetFrequency(frequency, channel))
+
+
+def shift_frequency(frequency: float,
+                    channel: channels.PulseChannel):
+    """Shift the ``frequency`` of a pulse ``channel``.
+
+    Examples:
+
+    .. code-block:: python
+        :emphasize-lines: 6
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.shift_frequency(1e9, d0)
+
+    Args:
+        frequency: Frequency in Hz to shift channel frequency by.
+        channel: Channel to shift frequency of.
+    """
+    raise NotImplementedError()
```

Comment:
Alternatively to one of my other comments, perhaps we can catch the not implemented error rather than expecting failure?

Reply:
This is just a placeholder. This should d not be merged with this method unimplemented.